src/broadcast.rs

```diff
@@ -89,19 +86,68 @@ pub struct Broadcast<NodeUid> {
     /// Whether we have already output a value.
     has_output: bool,
     /// The proofs we have received via `Echo` messages, by sender ID.
-    echos: BTreeMap<NodeUid, Proof<Vec<u8>>>,
+    echos: BTreeMap<N, Proof<Vec<u8>>>,
     /// The root hashes we received via `Ready` messages, by sender ID.
-    readys: BTreeMap<NodeUid, Vec<u8>>,
+    readys: BTreeMap<N, Vec<u8>>,
+    /// The outgoing message queue.
+    messages: VecDeque<TargetedMessage<BroadcastMessage, N>>,
+    /// The output, if any.
+    output: Option<Vec<u8>>,
 }
 
-impl<NodeUid: Eq + Hash + Debug + Clone + Ord> Broadcast<NodeUid> {
+impl<N: Eq + Debug + Clone + Ord> DistAlgorithm for Broadcast<N> {
+    type NodeUid = N;
+    type Input = Vec<u8>; // TODO: Allow anything serializable.
+    type Output = Self::Input;
+    type Message = BroadcastMessage;
+    type Error = Error;
+
+    fn input(&mut self, input: Self::Input) -> Result<(), Self::Error> {
+        if self.our_id != self.proposer_id {
+            return Err(Error::UnexpectedMessage);
```

Comment:
Possibly the error message name is obsolete. Whether the argument `input` arrives as part of a message depends on the context of application of the function `input`. Can the message name reflect why the input is rejected? E.g., `InstanceCannotPropose`?

Reply:
Done.